treeherder/client/thclient/client.py

```diff
@@ -552,6 +552,25 @@ def validate(self):
         for d in self.data:
             d.validate()
 
+    def get_chunks(self, chunk_size):
+        """
+        Return a list of new collections broken into chunks of size ``chunk_size``.
+
+        Each chunk will be a ``TreeherderCollection`` of the same
+        type as the original with a max of ``chunk_size`` count of
+        ``TreeherderData`` objects.
+
+        Each collection must then be POSTed individually.
+        """
+        artifact_collections = []
+        data_chunks = [self.data[o:o + chunk_size] for o
```

Comment:
Oops, I didn't realize that. Yeah let's stick to your solution :smiley: 
Maybe you can find a way to minimize the memory consumption returning a generator here?
Something like
```python
for i in range(0, len(self.data), chunk_size):
    yield self.__class__(self.data[i:i + chunk_size])
```